lisp/ess-r-flymake.el

```diff
@@ -0,0 +1,149 @@
+        :buffer (generate-new-buffer " *ess-r-flymake*")
+        :command `("R" "--slave" "--restore" "--no-save" "-e"
+                   ,(eval (concat
+                           "library(lintr);"
+                           ;; commandArgs(TRUE) lets us access
+                           ;; everything after --args as a string:
+                           "try(lint(commandArgs(TRUE)"
+                           ", " "linters = " ess-r-flymake-linters
+                           (when ess-r-flymake-lintr-cache ", cache = TRUE")
+                           "))"))
+                   "--args" ,(eval
+                              ;; text string of the current buffer:
+                              (concat (buffer-substring-no-properties
+                                       (point-min) (point-max)))))
+        :sentinel
+        (lambda (proc _event)
+          (when (eq 'exit (process-status proc))
+            (unwind-protect
+                ;; Only proceed if `proc' is the same as
+                ;; `ess-r--flymake-proc', which indicates that
+                ;; `proc' is not an obsolete process.
+                (if (with-current-buffer source (eq proc ess-r--flymake-proc))
+                    (with-current-buffer (process-buffer proc)
+                      (goto-char (point-min))
+                      ;; Parse the output buffer for diagnostic's
+                      ;; messages and locations, collect them in a list
+                      ;; of objects, and call `report-fn'.
+                      (cl-loop
+                       while (search-forward-regexp
+                              ;; Regex to match the output lint() gives us.
+                              (rx line-start "<text>:"
+                                  ;; row
+                                  (group-n 1 (one-or-more num)) ":"
+                                  ;; column
+                                  (group-n 2 (one-or-more num)) ": "
+                                  ;; type
+                                  (group-n 3 (| "style: " "warning: " "error: "))
+                                  ;; msg
+                                  (group-n 4 (one-or-more not-newline)) "\n")
+                              nil t)
+                       for msg = (match-string 4)
+                       for (beg . end) = (flymake-diag-region
+                                          source
+                                          (string-to-number (match-string 1))
+                                          (string-to-number (match-string 2)))
+                       for type = (let ((str (match-string 3)))
+                                    (cond ((string-equal str "error: ") :error)
+                                          ((string-equal str "warning: ") :warning)
+                                          ((string-equal str "style: ") :note)))
+                       collect (flymake-make-diagnostic source
```

Comment:
Why each argument on separate line? Looks rather ugly to me.

Reply:
How do you want it formatted?

Reply:
Frankly I couldn't care less how they do it. Most likely one of them copied from the other and that's not a good reason to imitate them. Having modularized code helps readability and debugging. Currently, if I want to debug that lambda I have to instrument all of the ess-r-flymake sausage.